tests/lib/WebServer.cc

```diff
@@ -417,9 +459,20 @@ WebServer::WebServer(const Pathname &root, unsigned int port, bool useSSL)
 {
 }
 
-void WebServer::start()
+TriBool WebServer::start()
 {
     _pimpl->start();
+
+    TriBool ret { checkLocalPort( _pimpl->port() ) };
+    if ( ret != true )
+    {
+      unsigned i = 0;
+      do {
+	sleep( 1 );
+	ret = checkLocalPort( _pimpl->port() );
+      } while ( ret != true || ++i == 10 );
+    }
+    return ret;
```

Comment:
Can we move that part into the worker_thread() main instead of the std::this_thread::sleep_for(std::chrono::milliseconds(100)); In case the port never opens its the correct place to clean up the webserver process and stop the thread right away. 

The start() should not return a TriBool it either started or not. WebServer::isStopped() already checks if it was started successful.

Reply:
done